packages/web-app-files/src/components/BatchActions.vue

```diff
@@ -1,118 +1,150 @@
+          {{ emptyTrashbinButtonText }}
+        </oc-button>
+      </li>
     </template>
-    <oc-grid v-if="displayBulkActions" gutter="small">
-      <div v-if="canDownload">
+    <template v-if="displayBulkActions">
+      <li v-if="canDownload">
         <oc-button
           id="download-selected-btn"
           key="download-selected-btn"
-          variation="primary"
+          :variation="isPrimaryVariation"
+          :appearance="isOutlineAppearance"
           @click="download"
         >
           <oc-icon name="archive" />
           <translate>Download</translate>
         </oc-button>
-      </div>
-      <div v-if="canCopy">
+      </li>
+      <li v-if="canCopy">
         <oc-button
           id="copy-selected-btn"
           key="copy-selected-btn"
-          variation="primary"
+          :variation="isPrimaryVariation"
+          :appearance="isOutlineAppearance"
           @click="triggerLocationPicker('copy')"
         >
           <oc-icon name="file_copy" />
           <translate>Copy</translate>
         </oc-button>
-      </div>
-      <div v-if="canMove">
+      </li>
+      <li v-if="canMove">
         <oc-button
           id="move-selected-btn"
           key="move-selected-btn"
-          variation="primary"
+          :variation="isPrimaryVariation"
+          :appearance="isOutlineAppearance"
           @click="triggerLocationPicker('move')"
         >
           <oc-icon name="folder-move" />
           <translate>Move</translate>
         </oc-button>
-      </div>
-      <div v-if="canDelete">
+      </li>
+      <li v-if="canDelete">
         <oc-button
           id="delete-selected-btn"
           key="delete-selected-btn"
-          variation="primary"
+          :variation="isPrimaryVariation"
+          :appearance="isOutlineAppearance"
           @click="$_deleteResources_displayDialog"
         >
           <oc-icon name="delete" />
           <translate>Delete</translate>
         </oc-button>
-      </div>
-      <div v-if="canAccept">
+      </li>
+      <li v-if="canAccept">
         <oc-button
           id="accept-selected-shares-btn"
           key="accept-shares-btn"
-          variation="primary"
+          :variation="isPrimaryVariation"
+          :appearance="isOutlineAppearance"
           @click="acceptShares()"
         >
           <oc-icon name="check" />
           <translate>Accept</translate>
         </oc-button>
-      </div>
-      <div v-if="canDecline">
+      </li>
+      <li v-if="canDecline">
         <oc-button
           id="decline-selected-shares-btn"
           key="decline-shares-btn"
-          variation="primary"
+          :variation="isPrimaryVariation"
+          :appearance="isOutlineAppearance"
           @click="declineShares()"
         >
           <oc-icon name="not_interested" />
           <translate>Decline</translate>
         </oc-button>
-      </div>
-    </oc-grid>
-  </div>
+      </li>
+      <li v-if="inContextActions && !isTrashbinRoute">
```

Comment:
You could use the `showDetails` action mixin here. You can copy the template from the ContextActions component and just use it here, it would avoid some code duplication.

Reply:
I initially did so, it turned out pretty ugly but let me take another look at it!

Reply:
Pretty sure this has been resolved by you by now, marking as resolved